src/mlpack/tests/ann/layer/separable_convolution.cpp

```diff
@@ -0,0 +1,91 @@
+/**
+ * @file tests/ann/layer/separable_convolution.cpp
+ * @author Aakash Kaushik
+ * @author Sidharth
+ *
+ * Tests the separable convolution layer modules.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <mlpack/core.hpp>
+#include <mlpack/methods/ann.hpp>
+
+#include "../../test_catch_tools.hpp"
+#include "../../catch.hpp"
+#include "../../serialization.hpp"
+#include "../ann_test_tools.hpp"
+
+using namespace mlpack;
+
+/**
+ * Simple Separable Convolution layer test.
```

Comment:
I think you haven't finished with the tests yet, but we should definitely have some more tests to ensure that the forward, backward, and gradient passes are correct.  It's likely you can adapt the existing convolution tests accordingly.  Please don't just run code from another a library and copy-paste the results; it's better to design or adapt an actual test.  If we hardcode values from another library, this is brittle: it's always possible their implementation is incorrect, or, it's also possible that they made some code choices that are not the only possible way to solve the problem.  (Think, e.g., small numerical precision errors and things of this nature.)

Reply:
sure i will note that and add some more tests to the file 👍

Reply:
I have added jacobian test and numerical gradient test, If you want to add any more tests do let me know